File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/statistics/StatisticsActivity.java
package io.github.sithengineer.motoqueiro.ui.statistics;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentTransaction;
import io.github.sithengineer.motoqueiro.BaseActivity;
import io.github.sithengineer.motoqueiro.R;

public class StatisticsActivity extends BaseActivity {

  public static Intent getNavigationIntent(Context context) {
    Intent intent = new Intent(context, StatisticsActivity.class);
    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
    return intent;
  }

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_content_frame);

    StatisticsFragment fragment =
        (StatisticsFragment) getSupportFragmentManager().findFragmentById(
            R.id.content_frame);

    Intent intent = getIntent();
    if (fragment == null) {
      fragment = StatisticsFragment.newInstance();

      FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
      transaction.addToBackStack("statistics_activity_fragment")
          .replace(R.id.content_frame, fragment)
          .commit();
    }
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/statistics/StatisticsActivity.kt
package io.github.sithengineer.motoqueiro.ui.statistics

import android.content.Context
import android.content.Intent
import android.os.Bundle
import io.github.sithengineer.motoqueiro.R
import io.github.sithengineer.motoqueiro.ui.BaseActivity

class StatisticsActivity : BaseActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_content_frame)

    val fragment = StatisticsFragment.newInstance()

    val transaction = supportFragmentManager.beginTransaction()
    transaction.addToBackStack("statistics_activity_fragment")
        .replace(R.id.content_frame, fragment)
        .commit()

  }

  companion object {

    fun getNavigationIntent(context: Context): Intent {
      val intent = Intent(context, StatisticsActivity::class.java)
      intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
      return intent
    }
  }
}
